runtime/lua/vim/lsp/rpc.lua

```diff
@@ -407,7 +407,10 @@ function Client:handle_body(body)
   end
   log.debug('rpc.receive', decoded)
 
-  if type(decoded.method) == 'string' and decoded.id then
+  if type(decoded) ~= 'table' then
+    -- Invalid server message
```

Comment:
```suggestion
```

The next line says `on_error(M.client_errors.INVALID_SERVER_MESSAGE` making this pretty obvious

Reply:
@mfussenegger FYI https://github.com/neovim/neovim/blob/37bf4c572a8fa20dc4a8433524ecc8c8d68fe53c/runtime/lua/vim/lsp/rpc.lua#L526-L527